src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -56629,5 +56629,371 @@ static void Main()
+
+            CompileAndVerify(piaCompilation, verify: VerifyOnMonoOrCoreClr);
+
+            string consumer = @"
+class UsePia
+{
+    public static void Main(ITest33.I1 x)
+    {
+    }
+} 
+";
+
+            foreach (var reference in new[] { piaCompilation.ToMetadataReference(embedInteropTypes: true), piaCompilation.EmitToImageReference(embedInteropTypes: true) })
+            {
+                var compilation1 = CreateCompilation(consumer, options: TestOptions.ReleaseDll, references: new[] { reference, attributesRef }, targetFramework: TargetFramework.NetStandardLatest);
+
+                compilation1.VerifyEmitDiagnostics(
+                    // (4,37): error CS1754: Type 'ITest33.I1' cannot be embedded because it is a nested type. Consider setting the 'Embed Interop Types' property to false.
+                    //     public static void Main(ITest33.I1 x)
+                    Diagnostic(ErrorCode.ERR_NoPIANestedType, "I1").WithArguments("ITest33.I1").WithLocation(4, 37)
+                    );
+            }
+        }
+
+        [Fact]
+        public void NoPia_07()
+        {
+            var attributes = CreateCompilation(NoPiaAttributes, options: TestOptions.ReleaseDll, targetFramework: TargetFramework.NetStandardLatest);
+            var attributesRef = attributes.EmitToImageReference();
+
+            string pia = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly: PrimaryInteropAssemblyAttribute(1,1)]
+[assembly: Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"")]
+
+[ComImport()]
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58279"")]
+public interface ITest33
+{
+    public static int F1;
+}
+";
+
+            var piaCompilation = CreateCompilation(pia, options: TestOptions.ReleaseDll, references: new[] { attributesRef }, targetFramework: TargetFramework.NetStandardLatest);
+
+            CompileAndVerify(piaCompilation, verify: VerifyOnMonoOrCoreClr);
+
+            string consumer = @"
+class UsePia
+{
+    public static void Main()
+    {
+        _ = ITest33.F1;
+    }
+} 
+";
+
+            foreach (var reference in new[] { piaCompilation.ToMetadataReference(embedInteropTypes: true), piaCompilation.EmitToImageReference(embedInteropTypes: true) })
+            {
+                var compilation1 = CreateCompilation(consumer, options: TestOptions.ReleaseDll, references: new[] { reference, attributesRef }, targetFramework: TargetFramework.NetStandardLatest);
+
+                compilation1.VerifyEmitDiagnostics(
+                    // (6,13): error CS8711: Type 'ITest33' cannot be embedded because it has a non-abstract member. Consider setting the 'Embed Interop Types' property to false.
+                    //         _ = ITest33.F1;
+                    Diagnostic(ErrorCode.ERR_DefaultInterfaceImplementationInNoPIAType, "ITest33").WithArguments("ITest33").WithLocation(6, 13)
+                    );
+            }
+        }
```

Comment:
Would it be interesting to test this combination:
ITest33 with method declaration. ITest34 inerits ITest33, implements the method. UsePia test with both ITest33 and ITest34.

Reply:
> Would it be interesting to test this combination:
ITest33 with method declaration. ITest34 inerits ITest33, implements the method. UsePia test with both ITest33 and ITest34.

Added tests.

---
In reply to: [266136460](https://github.com/dotnet/roslyn/pull/34171#discussion_r266136460) [](ancestors = 266136460)